core/jazz_environment-event-handler/index.js

```diff
@@ -207,9 +217,15 @@ function manageProcessItem(eventPayload, serviceDetails, configData, authToken)
 
     } else if (eventPayload.EVENT_NAME.S === configData.EVENTS.CREATE_BRANCH) {
       environmentApiPayload.friendly_name = svcContext.branch;
-      environmentApiPayload.deployment_descriptor = serviceDetails.deployment_descriptor
+      if (serviceDetails.deployment_descriptor) {
+        environmentApiPayload.deployment_descriptor = serviceDetails.deployment_descriptor
+      }
 
       exportable.processEventCreateBranch(environmentApiPayload, serviceDetails.id, configData, authToken)
+      .then((result) => { return exportable.createSafe(environmentApiPayload, serviceDetails.id, configData, authToken)})
```

Comment:
write one general method for all safe related activities and handle all these in that particular method rather than writing here.

Reply:
addressed as a wrapper function 'addSafe'